integration_tests/test_parser.py

```diff
@@ -15,64 +15,76 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 import os
+import fixtures
 import subprocess
 
 import integration_tests
+import testscenarios
 from snapcraft.tests import fixture_setup
 
 
-class TestParser(integration_tests.TestCase):
+class ParserTestCase(integration_tests.TestCase):
     """Test bin/snapcraft-parser"""
 
-    def test_parser_basic(self):
-        """Test snapcraft-parser basic usage"""
-        fixture = fixture_setup.FakePartsWiki()
-        self.useFixture(fixture)
+    def setUp(self):
+        super().setUp()
 
-        args = [self.snapcraft_parser_command, '--index',
-                fixture.fake_parts_wiki_fixture.url,
-                '--debug', '--output', 'parts.yaml']
-        subprocess.check_call(args, stderr=subprocess.DEVNULL,
-                              stdout=subprocess.DEVNULL)
+        tempdir = fixtures.TempDir()
+        self.useFixture(tempdir)
+        self.useFixture(fixtures.EnvironmentVariable('TMPDIR', tempdir.path))
+        self.tempdir = tempdir.path
 
-        self.assertTrue(os.path.exists('parts.yaml'))
+    def call_parser(self, wiki_path, expect_valid, expect_output=True):
+        part_file = os.path.join(self.tempdir, 'parts.yaml')
+        args = [self.snapcraft_parser_command, '--index', wiki_path,
+                '--debug',
+                '--output', part_file]
 
-    def test_hidden_snapcraft_yaml(self):
-        """Test hidden .snapcraft.yaml file."""
+        if expect_valid:
+            subprocess.check_call(args, stderr=subprocess.DEVNULL,
+                                  stdout=subprocess.DEVNULL)
+        else:
+            self.assertRaises(
+                subprocess.CalledProcessError,
+                subprocess.check_call, args, stderr=subprocess.DEVNULL,
+                stdout=subprocess.DEVNULL)
 
-        args = [self.snapcraft_parser_command, '--index',
-                os.path.join(os.path.dirname(__file__), 'hidden_parts_wiki'),
-                '--debug',
-                '--output', 'parts.yaml']
-        subprocess.check_call(args, stderr=subprocess.DEVNULL,
-                              stdout=subprocess.DEVNULL)
+        self.assertEqual(os.path.exists(part_file), expect_output)
```

Comment:
I think these tests should check the output content, not just if there exists any output. Of course, that's not something you have to fix here.
But, I wouldn't put the assertion inside the call. What do you think of tests like this:
```
parts_file_path = self.run_snapcraft_parser(...)
self.assertFileContents(parts_file_path, expectedContents)
```